connaisseur/key_store.py

```diff
@@ -60,9 +56,27 @@ def update(self, trust_data):
         """
 
         # update keys
-        keys = trust_data.get_keys()
-        for key in keys:
-            self.keys.setdefault(key, keys[key]["keyval"]["public"])
+        keys = dict(trust_data.get_keys())
+        signature_keys = [sig.get("keyid") for sig in trust_data.signatures]
```

Comment:
Would move this line below the comment as it only pertains to the very case described in the comment